jekyll/_cci2/language-java-maven.md

```diff
@@ -16,165 +16,334 @@ This guide will help you get started with a Java application building with Maven
+The Maven test runner with the [JaCoCo](https://www.eclemma.org/jacoco/) plugin
+generates a code coverage report during the build. To save that report as a
+build artifact, use the `store_artifacts` step.
 
-Next `store_test_results` uploads the test metadata from the `target/surefire-reports` directory so that it can show up in the CircleCI dashboard. 
+### A Configuration
+
+The following code sample is the entirety of a configuration file combining the features described above.
 
-Finally we store the uberjar as an [artifact](https://circleci.com/docs/2.0/artifacts/) using the `store_artifacts` step. From there this can be tied into a continuous deployment scheme of your choice.
 
 {% raw %}
+
 ```yaml
-...
-    steps:
+version: 2.0
 
+jobs:
+  test:
+    parallelism: 2
+    docker:
+      - image: circleci/openjdk:stretch
+    steps:
       - checkout
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "pom.xml" }}
+            - v1-dependencies-
+      - attach_workspace:
+          at: ./target
+      - run: |
+            ./mvnw \
+            -Dtest=$(for file in $(circleci tests glob "src/test/**/**.java" \
+            | circleci tests split --split-by=timings); \
+            do basename $file \
+            | sed -e "s/.java/,/"; \
+            done | tr -d '\r\n') \
+            -e test verify
+      - store_test_results:
+          path: target/surefire-reports
+      - store_artifacts:
+          path: target/site/jacoco/index.html
 
+  build:
+    docker:
+      - image: circleci/openjdk:stretch
+    steps:
+      - checkout
       - restore_cache:
-          key: circleci-demo-java-spring-{{ checksum "pom.xml" }}
-      
-      - run: mvn dependency:go-offline
-      
+          keys:
+            - v1-dependencies-{{ checksum "pom.xml" }}
+            - v1-dependencies-
+      - run: ./mvnw -Dmaven.test.skip=true package
       - save_cache:
           paths:
             - ~/.m2
-          key: circleci-demo-java-spring-{{ checksum "pom.xml" }}
-      
-      - run: mvn package
-      
-      - store_test_results:
-          path: target/surefire-reports
-      
-      - store_artifacts:
-          path: target/demo-java-spring-0.0.1-SNAPSHOT.jar
+          key: v1-dependencies-{{ checksum "pom.xml" }}
+      - persist_to_workspace:
+         root: ./
+         paths:
+           - target/
+
+workflows:
+  version: 2
+
+  build-then-test:
+    jobs:
+      - build
+      - test:
+          requires:
+            - build
 ```
 {% endraw %}
 
-Nice! You just set up CircleCI for a Java app using Maven and Spring.
+The configuration above is from a demo Java app, which you can access
+[here](https://github.com/CircleCI-Public/circleci-demo-java-spring). If you
+want to step through it yourself, you can fork the project on GitHub and
+download it to your machine. Go to the **Add Projects** page in CircleCI and click
```

Comment:
I think it's **Projects** and **Follow Projects** now? Not 100% sure on that though